sentry/src/db/channel.rs

```diff
@@ -67,3 +69,122 @@ pub async fn insert_channel(
         })
         .await
 }
+
+mod list_channels {
+    use crate::db::DbPool;
+    use bb8::RunError;
+    use bb8_postgres::tokio_postgres::types::{accepts, FromSql, ToSql, Type};
+    use chrono::{DateTime, Utc};
+    use primitives::{Channel, ValidatorId};
+    use std::error::Error;
+    use std::str::FromStr;
+
+    struct TotalCount(pub u64);
+    impl<'a> FromSql<'a> for TotalCount {
+        fn from_sql(ty: &Type, raw: &'a [u8]) -> Result<Self, Box<dyn Error + Sync + Send>> {
+            let str_slice = <&str as FromSql>::from_sql(ty, raw)?;
+
+            Ok(Self(u64::from_str(str_slice)?))
+        }
+
+        // Use a varchar or text, since otherwise `int8` fails deserialization
+        accepts!(VARCHAR, TEXT);
+    }
+
+    #[derive(Debug)]
+    pub struct ListChannels {
+        pub total_count: u64,
+        pub channels: Vec<Channel>,
+    }
```

Comment:
ListChannels is the same as `ChannelListResponse`

Reply:
One returns the `total_count`, while the other returns the `total_pages`, but we can move the calculation of the `total_pages` here and will change it to `ChannelListReponse`